core/drivers/stm32_rtc.c

```diff
@@ -430,47 +429,60 @@ static TEE_Result parse_dt(const void *fdt, int node)
 	if (res)
 		return res;
 
-	res = interrupt_dt_get(fdt, node, &rtc_dev.itr_chip,
-			       &rtc_dev.itr_num);
-	if (res != TEE_SUCCESS && res != TEE_ERROR_ITEM_NOT_FOUND)
-		return res;
+	if (rtc_dev.compat->has_rif_support)
+		cuint = fdt_getprop(fdt, node, "st,protreg", &lenp);
+
+	if (cuint) {
+		rtc_dev.conf_data = calloc(1, sizeof(*rtc_dev.conf_data));
+		if (!rtc_dev.conf_data)
+			panic();
 
-	if (fdt_getprop(fdt, node, "wakeup-source", NULL)) {
-		if (res != TEE_ERROR_ITEM_NOT_FOUND &&
-		    interrupt_can_set_wake(rtc_dev.itr_chip))
-			rtc_dev.rtc->is_wakeup_source = true;
-		else
-			DMSG("RTC wakeup source ignored");
+		rtc_dev.nb_res = (unsigned int)(lenp / sizeof(uint32_t));
+		assert(rtc_dev.nb_res <= RTC_NB_RIF_RESOURCES);
+
+		rtc_dev.conf_data->cid_confs = calloc(RTC_NB_RIF_RESOURCES,
+						      sizeof(uint32_t));
+		rtc_dev.conf_data->sec_conf = calloc(1, sizeof(uint32_t));
+		rtc_dev.conf_data->priv_conf = calloc(1, sizeof(uint32_t));
+		rtc_dev.conf_data->access_mask = calloc(1, sizeof(uint32_t));
+		if (!rtc_dev.conf_data->cid_confs ||
+		    !rtc_dev.conf_data->sec_conf ||
+		    !rtc_dev.conf_data->priv_conf ||
+		    !rtc_dev.conf_data->access_mask)
+			panic("Not enough memory capacity for RTC RIF config");
+
+		for (i = 0; i < rtc_dev.nb_res; i++)
+			stm32_rif_parse_cfg(fdt32_to_cpu(cuint[i]),
+					    rtc_dev.conf_data,
+					    RTC_NB_RIF_RESOURCES);
+
+		if (rtc_dev.conf_data->sec_conf[0] == RTC_RIF_FULL_SECURED)
+			rtc_dev.is_secured = true;
 	}
 
-	if (!rtc_dev.compat->has_rif_support)
+	cuint = fdt_getprop(fdt, node, "wakeup-source", NULL);
+	if (!cuint && !(IS_ENABLED(CFG_RTC_PTA) &&
+			IS_ENABLED(CFG_CORE_ASYNC_NOTIF) && rtc_dev.is_secured))
```

Comment:
Could you cross check this condition?
In which condition do you expect to handle the RTC interrupt?